### DISCUSSION TITLE
Retrieving page's title and favicon with JS Hooks

### DISCUSSION THREAD

**Discussion by JunoNgx** [pocketbase/pocketbase#4576] [Q&A] [answered]
Context: in my app, the user enters a random content, and if the content is detected to be a valid url, I want Pocketbase to try retrieving the title and absolute favicon url of this page.

```
try {
    // Processing to make sure it has a protocol
    const processedUrl = utils.tryProcessUrl(itemRecord.get("content"));
    const res = $http.send({
        url: processedUrl,
        method: "GET",
    });

    if (res.statusCode !== 200) {
        return;
    }

    console.log(JSON.stringify(res))
    const title = res.match(/<title[^>]*>([^<]+)<\/title>/);
    console.log(title)

} catch (err) {
    console.log(err)
}
```

I've managed to retrieve the html response content (which means I definitely have the data I need, I just need to scour through it), though it seems that `string` in Goja doesn't have `match()` for matching Regex expression. Jquery is also obviously out of the question.

I've managed to achieved this logic in Go extension with `net/http` and `goquery`, though I'd really like to stay in JS hooks (because PocketHost doesn't allow custom binary at the moment).

Any suggestion of how to rummage through the html content to find what I need, within the constraint of JS hooks?

**Comment by ganigeorgiev** [comment, marked as answer]
`match` is supported for strings in Goja but in your sample code `res` is an object, not a string.

If you want to access the `$http.send` response body as string, you can use `res.raw`.

Your particular case seems simple and a regex search should work fine, but if you are looking for an alternative you can try searching for external DOM parser library and load them with [require](https://pocketbase.io/docs/js-overview/#loading-modules) either as part of a `node_modules` folder or directly via its path (_any ES5 CJS module that doesn't use browser/node specific APIs should work_).

**Reply by JunoNgx** [reply]
You're definitely right on `res` not being a `string`. I forgot to import the types and missed that. `res.raw` is the correct property to look through. Thank you very much.

I haven't been able to write a regex pattern that has met my need despite having spent a long time; using a DOM parser might be a saner idea. Thank you for this suggestion, too; I'll look into it.
